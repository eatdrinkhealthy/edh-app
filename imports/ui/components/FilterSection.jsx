// @flow
import React, { Component } from "react";
import _ from "lodash";
import Pill from "./Pill";
import type { IFilter } from "../../state/reducers/filtersReducers";

class FilterSection extends Component {
  props: {
    title: string,
    filters: Array<IFilter>,
    getFilters: (Array<string>) => void,
  };

  state = {
    // state will be populated with id's of each filter
  };

  componentDidUpdate() {
    const setFilters = [];
    _.forIn(this.state, (value, key) => {
      if (value) setFilters.push(key);
    });

    this.props.getFilters(setFilters);
  }

  handleButtonClick = (event: SyntheticEvent<HTMLButtonElement>) => {
    // $FlowFixMe
    const { name } = event.currentTarget;

    this.setState({
      [name]: !this.state[name],
    });
  };

  render() {
    const { title, filters } = this.props;

    const divCenterContent = {
      textAlign: "center",
      padding: "0px 10px 0px 10px",
    };

    const pillContainer = {
      display: "inline-block",
    };

    return (
      <div>
        <div style={divCenterContent}>
          <span className="filterSectionTitle">{title}</span>
        </div>
        <div style={divCenterContent}>
          <span style={pillContainer}>
            <Pill
              name={filters[0].id}
              onClick={this.handleButtonClick}
              active={this.state[filters[0].id]}
            >
              {filters[0].name}
            </Pill>

            <Pill
              name={filters[1].id}
              onClick={this.handleButtonClick}
              active={this.state[filters[1].id]}
            >
              {filters[1].name}
            </Pill>

            <Pill
              name={filters[2].id}
              onClick={this.handleButtonClick}
              active={this.state[filters[2].id]}
            >
              {filters[2].name}
            </Pill>
          </span>
          <span style={pillContainer}>
            <Pill
              name={filters[3].id}
              onClick={this.handleButtonClick}
              active={this.state[filters[3].id]}
            >
              {filters[3].name}
            </Pill>

            <Pill
              name={filters[4].id}
              onClick={this.handleButtonClick}
              active={this.state[filters[4].id]}
            >
              {filters[4].name}
            </Pill>

            <Pill
              name={filters[5].id}
              onClick={this.handleButtonClick}
              active={this.state[filters[5].id]}
            >
              {filters[5].name}
            </Pill>
          </span>
        </div>
      </div>
    );
  }
}

export default FilterSection;
