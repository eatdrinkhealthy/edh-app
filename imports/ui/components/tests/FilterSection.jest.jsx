// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import FilterSection from "../FilterSection";
import type { IEatDrinkFilter } from "../../../state/reducers/eatDrinkFiltersReducers";

describe("<FilterSection />", function () {
  const testFilterList: Array<IEatDrinkFilter> = [
    { id: "juiceBar1", name: "Juice Bar1", on: true, foursquareCategory: "1" },
    { id: "juiceBar2", name: "Juice Bar2", on: false, foursquareCategory: "2" },
    { id: "juiceBar3", name: "Juice Bar3", on: false, foursquareCategory: "3" },
    { id: "juiceBar4", name: "Juice Bar4", on: true, foursquareCategory: "4" },
    { id: "juiceBar5", name: "Juice Bar5", on: false, foursquareCategory: "5" },
    { id: "juiceBar6", name: "Juice Bar6", on: false, foursquareCategory: "6" },
  ];

  it("matches render snapshot", function () {
    const tree = renderer.create(
      <FilterSection
        id="testFilter"
        title="Test Filter Section"
        filters={testFilterList}
        setFilter={() => {}}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call setFilter with filter id, when a Pill is clicked", function () {
    const props = {
      id: "testFilter",
      title: "Test Filter Section",
      filters: testFilterList,
      setFilter: jest.fn(),
    };

    const wrapper = mount(<FilterSection {...props} />);
    wrapper.find("[name='juiceBar1']").simulate("click");
    // NOTE: I have tried using 'toHaveBeenCalledWith' here, it was unreliable
    expect(props.setFilter.mock.calls[0][0]).toEqual("juiceBar1");
  });
});
