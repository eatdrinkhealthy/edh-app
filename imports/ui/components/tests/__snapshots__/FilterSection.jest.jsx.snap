// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSection /> matches render snapshot 1`] = `
<div
  id="testFilter"
>
  <div
    style={
      Object {
        "padding": "0px 10px 0px 10px",
        "textAlign": "center",
      }
    }
  >
    <span
      className="filterSectionTitle"
    >
      Test Filter Section
    </span>
  </div>
  <div
    style={
      Object {
        "padding": "0px 10px 0px 10px",
        "textAlign": "center",
      }
    }
  >
    <span
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <button
        className="pill pill_selected"
        name="juiceBar1"
        onClick={[Function]}
      >
        Juice Bar1
      </button>
      <button
        className="pill"
        name="juiceBar2"
        onClick={[Function]}
      >
        Juice Bar2
      </button>
      <button
        className="pill"
        name="juiceBar3"
        onClick={[Function]}
      >
        Juice Bar3
      </button>
    </span>
    <span
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <button
        className="pill pill_selected"
        name="juiceBar4"
        onClick={[Function]}
      >
        Juice Bar4
      </button>
      <button
        className="pill"
        name="juiceBar5"
        onClick={[Function]}
      >
        Juice Bar5
      </button>
      <button
        className="pill"
        name="juiceBar6"
        onClick={[Function]}
      >
        Juice Bar6
      </button>
    </span>
  </div>
</div>
`;
